src/Microsoft.Data/DataFrameColumnCollection.cs

```diff
@@ -4,47 +4,54 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 
 namespace Microsoft.Data
 {
     /// <summary>
     /// A DataFrameTable is just a container that holds a number of DataFrameColumns. It mainly acts as a convenient store to allow DataFrame to implement its algorithms
     /// </summary>
-    internal class DataFrameTable
+    public class DataFrameColumnCollection : Collection<DataFrameColumn>
     {
-        private IList<DataFrameColumn> _columns;
+        private Action ColumnsChanged;
 
         private List<string> _columnNames = new List<string>();
 
         private Dictionary<string, int> _columnNameToIndexDictionary = new Dictionary<string, int>(StringComparer.Ordinal);
 
         public long RowCount { get; internal set; }
 
-        public int ColumnCount { get; private set; }
-
-        public DataFrameTable()
+        public DataFrameColumnCollection(Action columnsChanged) : base()
         {
-            _columns = new List<DataFrameColumn>();
+            ColumnsChanged = columnsChanged;
         }
 
-        public DataFrameTable(IList<DataFrameColumn> columns)
+        public DataFrameColumnCollection(IList<DataFrameColumn> columns, Action columnsChanged) : base()
         {
             columns = columns ?? throw new ArgumentNullException(nameof(columns));
-            _columns = new List<DataFrameColumn>();
+            ColumnsChanged = columnsChanged;
             for (int i = 0; i < columns.Count; i++)
             {
-                InsertColumn(i, columns[i]);
+                Add(columns[i]);
             }
         }
 
-        public DataFrameColumn Column(int columnIndex) => _columns[columnIndex];
+        public IReadOnlyList<string> GetColumnNames()
+        {
+            var ret = new List<string>(Count);
+            for (int i = 0; i < Count; i++)
+            {
+                ret.Add(this[i].Name);
+            }
+            return ret;
+        }
 
         public IList<object> GetRow(long rowIndex)
```

Comment:
Why would we have this public API when it is on `DataFrame`?

Reply:
Removed.